examples/test_util/EclFile.hpp

```diff
@@ -16,26 +16,83 @@
    along with OPM.  If not, see <http://www.gnu.org/licenses/>.
    */
 
+#ifndef ECLFILE_HPP
+#define ECLFILE_HPP
+
+#include <opm/common/ErrorMacros.hpp>
+#include <examples/test_util/data/EclIOdata.hpp>
 
-//#include "EclIO.hpp"
 #include <iostream>
 #include <string>
 #include <fstream>
 #include <vector>
 #include <ctime>
 #include <map>
-
-#include <examples/test_util/data/EclIOdata.hpp>
+#include <unordered_map>
+#include <stdio.h>
 
 namespace EIOD = Opm::ecl;
 
-class EclFile 
+class EclFile
 {
-
 public:
-    
     explicit EclFile(const std::string& filename);
-    
-};  
+    bool formattedInput() { return formatted; }
+
+    void loadData();                            // load all data
+    void loadData(int arrIndex);                // load data based on array indices in vector arrIndex
+    void loadData(const std::vector<int>& arrIndex);   // load data based on array indices in vector arrIndex
+    void loadData(const std::string& arrName);         // load all arrays with array name equal to arrName
+
+    using EclEntry = std::tuple<std::string, EIOD::eclArrType, int>;
+    std::vector<EclEntry> getList() const;
+
+    template <typename T>
+    const std::vector<T>& get(int arrIndex) const;
+
+    template <typename T>
+    const std::vector<T>& get(const std::string& name) const;
+
+    bool hasKey(const std::string &name) const;
+
+protected:
+    bool formatted;
+    std::string inputFilename;
+
+    std::unordered_map<int, std::vector<int>> inte_array;
```

Comment:
Personally I think it is better policy to let calling scope carry the memory burden; with the current implementation it is not possible to reduse memory consumption after a keyword has been loaded. These filese can potentially be very large

Reply:
this would be a substantial rewrite with little gain. i have added a clearData() public member so you can clear out the data for now, then we can consider this for later.

Reply:
Sure - I agree it is a large rewrite.